test/diagonal.jl

```diff
@@ -463,7 +463,8 @@ Random.seed!(1)
         end
     end
 
-    @testset "svd (#11120/#11247)" begin
+    @testset "svd (#11120/#11247/#1149)" begin
+        D[1] = 0
```

Comment:
Is this changing `D` for all the tests coming after this? I see that tests pass, but perhaps it's better to introduce a small new testset with the example you gave in the original issue:

```julia
d = [1, 2, 3, 4, 0]
@test svd(diagm(d)) == svd(Diagonal(d))
```

?